velox/type/Timestamp.cpp

```diff
@@ -225,15 +229,33 @@ bool Timestamp::epochToUtc(int64_t epoch, std::tm& tm) {
   }
   tm.tm_year = y;
   tm.tm_yday = days;
-  auto* ip = monthLengths[leapYear];
-  for (tm.tm_mon = 0; days >= ip[tm.tm_mon]; ++tm.tm_mon) {
-    days = days - ip[tm.tm_mon];
-  }
-  tm.tm_mday = days + 1;
+  auto* months = daysBeforeFirstDayOfMonth[leapYear];
+  tm.tm_mon = std::upper_bound(months, months + 12, days) - months - 1;
+  tm.tm_mday = days - months[tm.tm_mon] + 1;
   tm.tm_isdst = 0;
   return true;
 }
 
+bool Timestamp::utcToEpoch(const std::tm& tm, int64_t& seconds) {
+  static_assert(sizeof(decltype(tm.tm_year)) == 4);
+  int64_t year = tm.tm_year + 1900ll;
```

Comment:
maybe add a comment on what is the 190011 magical number?

Reply:
never mind, just noticed it's as "ll" not "11". Making them uppercase would prevent this sort of confusion "LL"